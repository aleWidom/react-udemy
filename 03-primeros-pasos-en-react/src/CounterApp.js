import React, {useState} from 'react'
/* import PropTypes from 'prop-types' */

const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value); 

    const handleAdd = () => {
       setCounter(counter+1);
       //setCounter ((c) => c+1) se puede hacer así también.
    }

    const handleReset = () => (
        setCounter(value)
    )


    const handleSubstract = () => (
        setCounter(counter-1)
    )


    return (
        <>
        <h1>CounterApp</h1>
        <h2 aria-label='counter'>{counter}</h2>  
        <button onClick={handleAdd}>+1</button>
        <button aria-label='btn-reset' onClick= {handleReset}>Reset</button>
        <button onClick={handleSubstract}>-1</button>
        </>
    )
}

/* CounterApp.propTypes = {
    value: PropTypes.number
}



 */
export default CounterApp
