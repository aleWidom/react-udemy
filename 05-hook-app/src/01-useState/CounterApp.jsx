
import { useState } from "react"

export const CounterApp = () => {

    const [state, setState] = useState({
        counter1: 10,
        counter2: 20,
        counter3: 30
    }
    )

    const handleClick1 = () => {
        const newState = {
            ...state,
            counter1: state.counter1 + 1
        }
        setState(newState)
    }

    const handleClick2 = () => {
        const newState = {
            ...state,
            counter2: state.counter2 + 1
        }
        setState(newState)
    }

    const handleClick3 = () => {
        const newState = {
            ...state,
            counter3: state.counter3 + 1
        }
        setState(newState)
    }

    return (
        <>
            <h2>Counter: {state.counter1}</h2>
            <h2>Counter: {state.counter2}</h2>
            <h2>Counter: {state.counter3}</h2>
            <hr />
            <button onClick={handleClick1}>+1</button>
            <button onClick={handleClick2}>+1</button>
            <button onClick={handleClick3}>+1</button>

        </>
    )
}

