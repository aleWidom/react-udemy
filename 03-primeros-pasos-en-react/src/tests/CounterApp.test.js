import { render, screen, fireEvent } from "@testing-library/react"
import CounterApp from "../CounterApp"



describe('Pruebas en <CounterApp />', () => {

   const valor = 100;
   const initialValue = 10

   test('snapshot', () => {
      const { container } = render(<CounterApp />)
      expect(container).toMatchSnapshot()
   });

   test('debeRegresar100', () => {

      render(<CounterApp value={valor} />)
      expect(screen.getByText(valor)).toBeTruthy()
   });

   test('debeDeIncrementar+1', () => {
      //renderiza el componente digamos 
      render(<CounterApp value={initialValue} />)

      //el fireEvent simula eventos
      fireEvent.click(screen.getByText('+1'));
      //el screen es como la captura de lo que renderiza por lo que entendi
      expect(screen.getByText('11')).toBeTruthy()
   });

   test('debeDeDecrementar-1', () => {
      render(<CounterApp value={initialValue} />)
      //screen debug me deja ver como está el componenente en ese momento
      screen.debug()
      fireEvent.click(screen.getByText('-1'));
      //screen debug me deja ver como está el componenente en ese momento
      screen.debug()
      expect(screen.getByText('9')).toBeTruthy()
   });

   test('debeFuncionarReset', () => {
      render(<CounterApp value={initialValue} />)
      fireEvent.click(screen.getByText('+1'));
      fireEvent.click(screen.getByText('+1'));
      fireEvent.click(screen.getByText('+1'));
/*       fireEvent.click(screen.getByText('Reset')); */

      //para mi una mejor forma para que no ocurra errores a futuro
      /* screen.getByRole busco a través de etiqueta */
      fireEvent.click(screen.getByRole('button', {name:'btn-reset'} ))
      expect(screen.getByRole('heading', {level:2}, {name:'counter'} )).toBeTruthy()
      expect(screen.getByRole('heading', {level:2}, {name:'counter'} ).innerHTML).toContain(initialValue)
   });

})
