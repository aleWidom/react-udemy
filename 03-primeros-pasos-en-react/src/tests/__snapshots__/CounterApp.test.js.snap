// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    aria-label="counter"
  />
  <button>
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button>
    -1
  </button>
</div>
`;
